src/test/java/com/meetme/plugins/jira/gerrit/adminui/AdminServletTest.java

```diff
@@ -0,0 +1,172 @@
+            }
+        });
+    }
+
+    /**
+     * Initialise the mock-REST-API, call
+     * {@link com.sonymobile.tools.gerrit.gerritevents.GerritQueryHandlerHttp#queryJava(String, boolean, boolean, boolean)}.
+     */
+    @Before
+    public void setupMockServer() {
+        wireMockServer = new WireMockServer(new WireMockConfiguration().port(port));
+        //.usingFilesUnderDirectory("src/test/"));
+        wireMockServer.start();
+    }
+
+    /**
+     * Shut down the mock-REST-API.
+     */
+    @After
+    public void shutdownMockServer() {
+        wireMockServer.stop();
+    }
+
+    /**
+     * setup stub for wireMockServer.
+     * @param expectedCode the statuscode
+     */
+    public void setupStubForRequests(int expectedCode) {
+        ///a/changes/?q=limit%3A1
+        wireMockServer.stubFor(get(urlEqualTo(query))
+                .withBasicAuth(username, password)
+                .willReturn(aResponse().withHeader("Content-Type", "text/plain")
+                        .withStatus(expectedCode)
+                        .withBody("\n []")));
+    }
+
+
+    /**
+     * test {@link AdminServlet#performConnectionTest(GerritConfiguration, Map)}.
+     * As is, only connections to the REST-endpoint are tested
+     */
+    @Test
+    public void testPerformConnectionTest() {
+        //setting all required fields
+        gerritConfiguration.setPreferRestConnection(true);
+        gerritConfiguration.setHttpBaseUrl(url);
+        gerritConfiguration.setHttpUsername(username);
+        gerritConfiguration.setHttpPassword(password);
+
+        setupStubForRequests(200);
+        adminServlet.performConnectionTest(gerritConfiguration, map);
+        assertTrue((Boolean) map.get("testResult"));
+
+        setupStubForRequests(400);
+        adminServlet.performConnectionTest(gerritConfiguration, map);
+        assertFalse((Boolean) map.get("testResult"));
+        assertTrue(((String) map.get("testError")).endsWith("(400)"));
+
+        setupStubForRequests(401);
+        adminServlet.performConnectionTest(gerritConfiguration, map);
+        assertFalse((Boolean) map.get("testResult"));
+        assertTrue(((String) map.get("testError")).endsWith("(401)"));
+        System.out.println(map.toString());
+
+        setupStubForRequests(404);
+        adminServlet.performConnectionTest(gerritConfiguration, map);
+        assertFalse((Boolean) map.get("testResult"));
+        assertTrue(((String) map.get("testError")).endsWith("(404)"));
+        System.out.println(map.toString());
+
+        //testing other response code
+        setupStubForRequests(418);
+        adminServlet.performConnectionTest(gerritConfiguration, map);
+        assertFalse((Boolean) map.get("testResult"));
+        assertTrue(((String) map.get("testError")).endsWith("(418)"));
+        System.out.println(map.toString());
+    }
+
+
+    /**
+     * Useful for testing against real servers
+     */
+    //@Test
+    public void testArbitraryServer()
+    {
```

Comment:
könntest du mal schauen, ob du den Code durchgängig gleich formatiert bekommst? ggf. auch in Richtung "wie er im Original" formatiert ist, ansonsten würde es noch mehr Änderungen geben.